assets/js/OutputIframe.js

```diff
@@ -1,28 +1,120 @@
 import React from 'react'
 import IframeCtrl from './IframeCtrl'
-
+import {Modal, ModalBody, ModalHeader} from 'reactstrap'
+import {store, setWorkflowPublicAction} from "./workflow-reducer";
 let OutputIframeCtrl;
 
 class OutputIframe extends React.Component {
   constructor(props) {
     super(props);
+    this.toggleModal = this.toggleModal.bind(this);
+    this.toggleSetPublicModal = this.toggleSetPublicModal.bind(this);
+    this.toggleEmbedIframeModal = this.toggleEmbedIframeModal.bind(this);
+    this.setPublic = this.setPublic.bind(this);
     this.state = {
-      src: ("/api/wfmodules/" + this.props.selectedWfModuleId + "/output?rev=" + this.props.revision)
+      src: ("/api/wfmodules/" + this.props.selectedWfModuleId + "/output?rev=" + this.props.revision),
+      setPublicModalOpen: false,
+      embedIframeModalOpen: false
     }
   }
 
   componentWillReceiveProps(nextProps) {
     this.setState({
       src: ("/api/wfmodules/" + nextProps.selectedWfModuleId + "/output?rev=" + nextProps.revision)
-    })
+    });
+    if (this.state.setPublicModalOpen
+      && nextProps.workflow.public
+      && (nextProps.workflow.public !== this.props.workflow.public)) {
+      this.toggleSetPublicModal();
+      this.toggleEmbedIframeModal();
+    }
   }
 
   setRef(ref) {
     OutputIframeCtrl = new IframeCtrl(ref);
   }
 
+  toggleSetPublicModal() {
+    this.setState({
+      setPublicModalOpen: !this.state.setPublicModalOpen
+    })
+  }
+
+  toggleEmbedIframeModal() {
+    this.setState({
+      embedIframeModalOpen: !this.state.embedIframeModalOpen
+    });
+  }
+
+  toggleModal() {
+    if (!this.props.workflow.public) {
+      this.toggleSetPublicModal();
+    } else {
+      this.toggleEmbedIframeModal();
+    }
+  }
+
+  setPublic() {
+    store.dispatch(setWorkflowPublicAction(this.props.workflow.id, true));
+  }
+
+  renderSetPublicModal() {
+    return (
+      <Modal isOpen={this.state.setPublicModalOpen} toggle={this.toggleModals} className='test-setpublic-modal'>
```

Comment:
Is this the same code as the public modal in (IIRC) WorkflowMetadata? Should be a shared component if so.

Reply:
Sort of, but that works because WorkflowMetadata and NavBar share state. The entire handling of public and readOnly needs refactoring using redux. Then we can make this shared code.